andela/ah-frontend-prime
# 164798252 users should be able to reset their passwords

### What does this PR do?

-  Enable users to be able to reset their password in case they have forgotten

### Description of tasks to be completed.

- Add tests for a password reset  functionality
- send a link to  the user's email with their password reset token
- Add pages send an email and resetting password respectively

### How should this be manually tested?

Visit the deployed application page on heroku
enter your email that you had registered with in the app,then you will recieve an email with a password reset link, follow this link that will redirect to the password reset page from where you enter and confirm your new password.

### Any background context you want to provide?

    N/A

What are the relevant pivotal tracker stories?
[#164798252](https://www.pivotaltracker.com/story/show/164798252)

### Screen shots
send password resetlink page
<img width="1182" alt="Screenshot 2019-05-16 at 19 45 23" src="https://user-images.githubusercontent.com/18057325/57872431-573ff880-7814-11e9-9d17-9db008fda1b9.png">

<img width="1281" alt="Screenshot 2019-05-16 at 19 51 58" src="https://user-images.githubusercontent.com/18057325/57872482-73dc3080-7814-11e9-8755-be155fa26a30.png">

password reset page
<img width="1137" alt="Screenshot 2019-05-16 at 19 52 54" src="https://user-images.githubusercontent.com/18057325/57872589-ae45cd80-7814-11e9-979a-3ffd2e64f037.png">
